File: AAAA/dominique/C/TPNOTE/facile2.pl


author= DR
extends=/ComputerScience/python/AP1-1920/templates/pltest.pl


title = Evaluation d'une fonction en un point

text==

Soit la fonction de $%\mathbb{R} \in \mathbb{R}%$ définie par $%f(x) = ax^2+bx+c %$.

Ecrire une fonction **eval_f(x,a,b,c)** qui retourne la valeur de f en x.


==



before==
import random
for i in range(5):
    if i==0:
        a,b,c,x= 1,1,1,1
    elif i% 2==1:
        a=random.random()
        b=random.random()
        c=random.random()
        x=round(random.random()*3,2)
    else:
        a=random.randint(1,5)
        b=random.randint(-5,5)
        c=random.randint(-5,5)
        x=random.random()
    globals()[f"pltest{i}"]= f""">>> eval_f({x},{a},{b},{c}) 
{ a*x*x+b*x+c}
"""

==

        

File: ComputerScience/C/TPNOTE/facile2.c


// PL:author=DR
/* PL:title= Facile  */
/* PL:text==

Ecrire une fonction **eval_f** qui prend en paramètre un flottant et 3 entiers (dans cet ordre) et qui retourne un flottant.
 

Votre fonction **eval_f** calcule la valeur de la fonction $%f(x)={latexcode}%$.

PL:== */


// PL:code_before==
#include <stdio.h>
#include <stdlib.h>

#ifndef PCODE
#define PCODE 0.0
#endif
// PL:==
// PL:solution==

float eval_f(float x, int a , int b , int c){
    return PCODE;
}

// PL:==

/* PL:code==
void eval_f(){
    
printf("Plus longue partie constante %d\n",...);
printf("Plus longue partie croissante %d\n",...);

}

PL:== */

// PL:code_after==


void randomtest(){
    int e,f,g;
    float j;
    e =rand() % 100;
    f =rand() % 100;
    g =rand() % 10;
    j = 2.0 * (rand()% 100);
    printf("eval_f(%.2f,%d,%d,%d) = %.2f\n",j,e,f,g,eval_f(j,e,f,g));
}


int main(int argc, char **v){
    if (argc < 2){
        printf("usage %s seed\n",v[0]);
        exit(1);
    }
    srand(atoi(v[1]));
    printf("eval_f(1.0,1,-1,1) = %.2f\n",eval_f(1.0,1,-1,1));
    printf("eval_f(2.0,2,-1,-1) = %.2f\n",eval_f(2.0,2,-1,-1));
    randomtest();
    randomtest();
    randomtest();
    randomtest();
    return 0;
}


// PL:==
/* PL:aleacode==

a=2
# this python code is executed in the buider to generate the code 
# to be compared to the student code
a=1
import random 

# expression en python et en latex 
posibilities = [("x*(a*a+b+c)","x*(a^2+b+c)")]
# choix d'une expression
pcode,latexcode = random.choice(posibilities)

solution = f"""#define PCODE {pcode}\n"""+solution

PL:== */



/* PL:checks_args_stdin==
[["Test 1", ["11"], ""],
["Test Yveline", ["78"], ""],
]
PL:== */